modules-js/react-fleet/src/components/CompatibilityWarning.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { detect, parseUserAgent } from 'detect-browser';
+
+export enum Message {
+  NO_SCRIPT,
+  OLD_BROWSER,
+}
+
+export interface Props {
+  userAgent: string;
+}
+
+/**
+ * Place this component on the page to show browser compatibility warnings.
+ *
+ * This component needs to be able to run purely server-side, with any
+ * client-side JS handled outside of React. That’s because old browsers
+ * potentially won’t be able to start up the app JS at all.
+ *
+ * If you’re using Next, a good place for this is just after <Main /> in
+ * _document.tsx.
+ */
+export default function CompatibilityWarning({ userAgent }: Props) {
```

Comment:
This is an awesome utility component to have in our library! Great idea. 💯

Reply:
Thanks!